next-cmake/CMakeLists.txt

```
@@ -0,0 +1,464 @@
+        CACHE PATH "path to tensile library")
+  else()
+    set(ROCBLAS_TENSILE_LIBRARY_DIR
+        "\${CPACK_PACKAGING_INSTALL_PREFIX}${CMAKE_INSTALL_LIBDIR}/rocblas/library"
+        CACHE PATH "path to tensile library")
+  endif()
+  set(ROCBLAS_TENSILE_SUBDIR_PATH
+      "${CMAKE_CURRENT_SOURCE_DIR}/../../Tensile"
+      CACHE STRING "Path to Tensile subdirectory")
+endif()
+
+if(ROCBLAS_ENABLE_HOST)
+  set(ROCBLAS_BUILD_SHARED_LIBS
+      ON
+      CACHE BOOL "Build the rocBLAS shared or static library")
+  set(ROCBLAS_ENABLE_OPENMP
+      ON
+      CACHE BOOL "Enable OpenMP support.")
+  # I don't know that we can build with BLIS OFF
+  set(ROCBLAS_ENABLE_BLIS
+      ON
+      CACHE BOOL "Enable BLIS support.")
+  set(ROCBLAS_ENABLE_TENSILE
+      ON
+      CACHE BOOL "Build rocBLAS host library with Tensile backend.")
+  set(ROCBLAS_ENABLE_HIPBLASLT
+      ON
+      CACHE BOOL "Build rocBLAS host library with hipBLASLt backend.")
+  set(ROCBLAS_CONFIG_DIR
+      "\${CPACK_PACKAGING_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}"
+      CACHE PATH "Path placed into ldconfig file")
+  query_git_hash(rocblas ${CMAKE_CURRENT_SOURCE_DIR})
+endif()
+
+if(ROCBLAS_ENABLE_CLIENT)
+  set(ROCBLAS_ENABLE_BENCHMARKS
+      ON
+      CACHE BOOL "Build benchmark client.")
+  set(ROCBLAS_ENABLE_TESTS
+      ON
+      CACHE BOOL "Build test client.")
+  set(ROCBLAS_ENABLE_SAMPLES
+      ON
+      CACHE BOOL "Build client samples.")
+  set(ROCBLAS_ENABLE_FORTRAN
+      ON
+      CACHE BOOL "Build Fortran clients.")
+  # rocm-smi is not presently available on Windows so we do not require it.
+  cmake_dependent_option(ROCBLAS_ENABLE_ROCM_SMI "Require rocm_smi." ON
+                         "NOT WIN32" OFF)
+endif()
+
+set(ROCBLAS_ENABLE_ASAN
+    OFF
+    CACHE BOOL "Build with address sanitizer enabled.")
+set(ROCBLAS_ENABLE_COVERAGE
+    OFF
+    CACHE BOOL "Build with gcov support.")
+
+find_package(hip REQUIRED)
+
+if(ROCBLAS_ENABLE_BLIS)
+  find_package(BLIS REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_OPENMP)
+  find_package(OpenMP REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_HIPBLASLT)
+  if(NOT TARGET roc::hipblaslt)
+    find_package(hipblaslt 1.0.0 REQUIRED)
+  endif()
+  rocm_package_add_dependencies(DEPENDS "hipblaslt >= 1.0.0")
+endif()
+
+if(ROCBLAS_ENABLE_ROCM_SMI)
+  find_package(rocm_smi REQUIRED)
+endif()
+
+if(ROCBLAS_ENABLE_DEVICE)
+  message(STATUS "Configuring device libraries...")
+  if(NOT TARGET roc::tensile-host)
+    message(STATUS "Adding Tensile from subdirectory: ${ROCBLAS_TENSILE_SUBDIR_PATH}")
+    add_subdirectory(${ROCBLAS_TENSILE_SUBDIR_PATH}/next-cmake
+                     ${CMAKE_CURRENT_BINARY_DIR}/Tensile)
+  endif()
+endif()
+
+if(ROCBLAS_ENABLE_HOST)
+  message(STATUS "Configuring host library")
+  add_subdirectory(host-library)
+  target_link_libraries(rocblas PRIVATE roc::tensile-host)
+endif()
+
+if(ROCBLAS_ENABLE_CLIENT)
+  find_package(GTest REQUIRED)
+  find_package(cblas REQUIRED)
+  find_package(BLAS REQUIRED)
```

```suggestion
  find_package(BLAS REQUIRED)
  find_package(cblas REQUIRED)
```
Order could have an impact.

---

I will reorder this.